src/EnergyPlus/GlobalNames.cc

```diff
@@ -75,11 +75,11 @@ void IntraObjUniquenessCheck(EnergyPlusData &state,
     if (NameToVerify.empty()) {
         ShowSevereError(state, format("E+ object type {} cannot have a blank {} field", CurrentModuleObject, FieldName));
         ErrorsFound = true;
-        NameToVerify = "xxxxx";
+        // NameToVerify = "xxxxx";
```

Comment:
Names are never empty now (epJSON catches that) so this is not necessary which allows us to make the input const which helps in several other ways.